lib/modules/datasource/hermit/index.ts

```diff
@@ -0,0 +1,144 @@
+ */
+export class HermitDatasource extends Datasource {
+  static readonly id = 'hermit';
+
+  override readonly customRegistrySupport = true;
+
+  override readonly registryStrategy = 'first';
+
+  override readonly defaultVersioning = id;
+
+  override readonly defaultRegistryUrls = [
+    'https://github.com/cashapp/hermit-packages',
+  ];
+
+  pathRegex: RegExp;
+
+  constructor() {
+    super(HermitDatasource.id);
+    this.http = new GithubHttp(GithubReleasesDatasource.id);
+    this.pathRegex = regEx('^\\/(?<owner>[^/]+)\\/(?<repo>[^/]+)$');
+  }
+
+  @cache({
+    namespace: `datasource-hermit-package`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl ?? ''}-${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    logger.trace(`HermitDataSource.getReleases()`);
+
+    if (!registryUrl) {
+      logger.error('registryUrl must be supplied');
+      return null;
+    }
+
+    if (!registryUrl.startsWith('https://github.com/')) {
+      logger.error({ registryUrl }, 'Only Github registryUrl is supported');
+      return null;
+    }
+
+    const items = await this.getHermitSearchManifest(registryUrl);
+
+    if (items === null) {
+      return null;
+    }
+
+    const res = items.find((i) => i.Name === packageName);
+
+    if (!res) {
+      logger.error({ packageName, registryUrl }, 'cannot find hermit package');
+      return null;
+    }
+
+    const sourceUrl = res.Repository;
+
+    return {
+      sourceUrl,
+      releases: [
+        ...res.Versions.map((v) => ({
+          version: v,
+          sourceUrl,
+        })),
+        ...res.Channels.map((v) => ({
+          version: v,
+          sourceUrl,
+        })),
+      ],
+    };
+  }
+
+  /**
+   * getHermitSearchManifest fetch the index.json from release
+   * named index, parses it and returned the parsed JSON result
+   */
+  @cache({
+    namespace: `datasource-hermit-search-manifest`,
+    key: ({ registryUrl }: GetReleasesConfig) => registryUrl ?? '',
+  })
+  async getHermitSearchManifest(
+    registryUrl: string
+  ): Promise<HermitSearchResult[] | null> {
+    const u = new URL(registryUrl);
+    const host = u.host;
+    const groups = this.pathRegex.exec(u.pathname)?.groups;
+
+    if (!groups) {
+      return null;
+    }
+
+    const { owner, repo } = groups;
+
+    const apiBaseUrl = getApiBaseUrl(`https://${host}`);
+
+    const indexRelease = await this.http.getJson<GithubRelease>(
+      `${apiBaseUrl}repos/${owner}/${repo}/releases/tags/index`
+    );
+
+    const asset = indexRelease.body.assets.find(
+      (asset: GithubReleaseAsset) => asset.name === 'index.json'
+    );
+
+    if (!asset) {
+      logger.error(
+        { registryUrl },
+        `can't find asset index.json in the given registryUrl`
+      );
+      return null;
+    }
+
+    const indexContent = await streamToString(
+      this.http.stream(asset.url, {
```

Comment:
did you tried to use `getJson` function?

Reply:
The very first implementation I use `http.get` to get the search manifest. But you didn't like it back then and told me to use `http.stream` like the examples from `terraform` and `bazel`. I did and I made the change on #16255 so it works with the current implementation. Can we please be consistent on the same topic?

https://github.com/renovatebot/renovate/pull/16255#discussion_r907136376

Reply:
not sure it might happen here but i've seen this happen before, where URL doesnt return JSON but returns 404 in an HTML error
there's no handling for the `JSON.parse(indexContent)` here in case it fails parsing or returns some HTML instead of a json..

so i think to use what @viceice mentioned above is better

Reply:
@viceice was very not happy and asked to use `http.stream` instead (see [this comment](https://github.com/renovatebot/renovate/pull/16255#discussion_r907136376)), which I did.  I would like not to make any undesired changes unless a clear and acceptable reason given.

Reply:
It was because before it wasn't clear to me, that we download `JSON` here. So why we need the `accept` header?
Shouldn't the default `getJson` work? which should send a `accept: application/json` and return the already parsed string as object.

Reply:
We need to use the `accept` header `application/octet-stream` because it will be needed to download Release Asset for private Github Repository. See [Github Release Asset Document](https://docs.github.com/en/rest/releases/assets).  

I have [explained](https://github.com/renovatebot/renovate/pull/16257#discussion_r914470336) before in our case we are downloading release assets from private Github Repository instead of the open source one. 

Hitting the URL with `getJSON` won't work because hitting the URL with `getJSON` will only return the content of the API call, not the content of the manifest.  like the example screenshot from Github's document.

![image](https://user-images.githubusercontent.com/1623058/178234465-6c5483f8-2e64-4b9d-968e-6260bc76536d.png)

Reply:
if you need the manifest then just handle the JSON parsing with try/catch or throw error depending on your logic and it should be ok i guess

Reply:
Alright, this sounds much more reasonable. I'll wrap the `JSON.parse` with try catch and return `null` instead, which fits the logic in the invoking function.  Thanks @PhilipAbed

Reply:
and also please add some code comment, why you need to use `accept` header and streaming here

Reply:
@PhilipAbed this is done as discussed, PTAL.